go/app/server.go

```diff
@@ -147,19 +172,55 @@ func (s *Handlers) AddItem(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// GetItems is a handler to return resistered items
+func (s *Handlers) GetItems(w http.ResponseWriter, r *http.Request) {
+    ctx := r.Context()
+
+    // データ取得
+    items, err := s.itemRepo.GetAll(ctx)
+    if err != nil {
+        http.Error(w, "failed to retrieve items", http.StatusInternalServerError)
+        return
+    }
+
+    // JSONレスポンスを返す
+    resp := struct {
+        Items []Item `json:"items"`
+    }{Items: items}
+
+    json.NewEncoder(w).Encode(resp)
+}
+
+
 // storeImage stores an image and returns the file path and an error if any.
 // this method calculates the hash sum of the image as a file name to avoid the duplication of a same file
 // and stores it in the image directory.
 func (s *Handlers) storeImage(image []byte) (filePath string, err error) {
 	// STEP 4-4: add an implementation to store an image
+
 	// TODO:
 	// - calc hash sum
+	hash := sha256.Sum256(image)
+	fileName := fmt.Sprintf("%x.jpg",hash)
+
 	// - build image file path
+	filePath = filepath.Join(s.imgDirPath, fileName)
+
 	// - check if the image already exists
+	if _, err = os.Stat(filePath); err == nil{
+		return fileName, nil
+	}
+
 	// - store image
+	fmt.Println("Savin new image:",fileName)
```

Comment:
[nits]
他の場所で、ログ出力は `slog` を用いているので、それに倣った方が良さそうです
他の場所も同様です :eyes:

```suggestion
	slog.Info("Saving new image:", "fileName", fileName)
```